src/GraphNavigator/DeserializationGraphNavigator.php

```diff
@@ -209,13 +209,20 @@ public function accept($data, ?array $type = null)
                         continue;
                     }
 
-                    $this->context->pushPropertyMetadata($propertyMetadata);
+                    /** Metadata changes based on context, should not be cached  */
+                    $contextSpecificMetadata = $propertyMetadata;
+                    if (null !== $this->context->getPropertyNamingStrategy()) {
```

Comment:
@goetas I know you mentioned that this may be slow. 
This is optional however, as `Serializer Builder` exposes `default` property naming strategy:

```php
->setPropertyNamingStrategy()
```

So this will be only used conditionally in case someone demand this. 
Will this be fine?